src/opengb/dram/quantity.cljc

```diff
@@ -61,3 +81,106 @@
 (s/def ::mass-per-year (s/and quantity? #(mass-per-year-unit? (get-unit %))))
 
 (s/def ::mass-intensity (s/and quantity? #(mass-intensity-unit? (get-unit %))))
+
+(s/def ::volume-intensity (s/and quantity? #(volume-intensity-unit? (get-unit %))))
+
+(s/def ::us-customary (s/and quantity? #(us-customary-unit? (get-unit %))))
+
+(s/def ::metric (s/and quantity? #(metric-unit? (get-unit %))))
+
+;; * Conversions
+
+;; ** Making Quantities
+
+(defn can-convert-to-quantity?
+  "True when the given `magnitude` and `unit` can be converted into a quantity."
+  [magnitude unit]
+  (and (s/valid? ::magnitude magnitude)
+       (s/valid? ::unit unit)))
+
+;; ** Intensities and Totals
+
+(defn calculate-total-water-use
+  "Provides total water use in l/year."
+  [intensity area]
+  {:pre [(s/valid? ::volume-intensity intensity)
+         (s/valid? ::metric intensity)
+         (s/valid? ::area area)
+         (s/valid? ::metric area)]}
+  (let [water-magnitude   (get-magnitude intensity)
+        area-magnitude    (get-magnitude area)
+        total-volume-unit "l/year"]
+    (make-quantity (* water-magnitude
+                      area-magnitude)
+                   total-volume-unit)))
+
+(defn calculate-total-energy-use
+  "Calculates total energy use in kWh/year."
+  [intensity area]
+  {:pre [(s/valid? ::energy-use-intensity intensity)
+         (s/valid? ::metric intensity)
+         (s/valid? ::area area)
+         (s/valid? ::metric area)]}
+  (let [energy-magnitude  (get-magnitude intensity)
+        area-magnitude    (get-magnitude area)
+        total-energy-unit "kWh/year"]
+    (make-quantity (* energy-magnitude
+                      area-magnitude)
+                   total-energy-unit)))
+
+(defn intensity->total
+  [intensity area]
+  (cond
+    (s/valid? ::energy-use-intensity intensity) (calculate-total-energy-use intensity area)
+    (s/valid? ::volume-intensity intensity)     (calculate-total-water-use intensity area)
+    :default                                    (throw (ex-info "Only supported for volume and energy intensities." {:intensity intensity
+                                                                                                                     :area      area}))))
+
+;; ** US Customary and Metric
+
+(defn energy-intensity-to-metric
+  "Converts a US Customary energy intensity to the Metric equivalent."
```

Comment:
Let's pull this out into one big lookup around the magnitudes, something like

`{
["kWh/m**2/year" "kBtu/ft**2/year"] #(* 3.155 %)
,,,
}`

and one outer `us-customary->metric` function.

(This is all a stopgap until we figure out how to rely on a definition file like https://github.com/opengb/dram/blob/registry/resources/default_en.txt, which is my ultimate goal here. It may take a bit of time to get there, so we'll probably want to ad-hoc add some more conversions without it being a full new fn with error handling, etc. over and over).